Shared/Samples/Navigate route/NavigateRouteView.swift

```diff
@@ -0,0 +1,351 @@
+                
+                Spacer()
+                
+                Button("Recenter") {
+                    model.autoPanMode = .navigation
+                }
+                .disabled(!model.isNavigatingRoute || model.autoPanMode == .navigation)
+                
+                Spacer()
+                
+                Button("Reset") {
+                    Task {
+                        await model.resetNavigation()
+                        model.isNavigatingRoute = false
+                    }
+                }
+                .disabled(model.isResettingRoute)
+            }
+        }
+    }
+}
+
+private extension NavigateRouteView {
+    /// A view model for this sample.
+    @MainActor
+    class Model: ObservableObject {
+        /// A Boolean value indicating whether the navigate button is disabled.
+        @Published var isNavigateDisabled = true
+        
+        /// A Boolean value indicating whether the sample is navigating the route.
+        @Published var isNavigatingRoute = false
+        
+        /// A Boolean value indicating whether navigation is being reset.
+        @Published var isResettingRoute = false
+        
+        /// The error shown in the alert.
+        @Published var error: Error?
+        
+        /// The viewpoint of the map.
+        @Published var viewpoint: Viewpoint?
+        
+        /// The current auto-pan mode.
+        @Published var autoPanMode: LocationDisplay.AutoPanMode {
+            didSet {
+                locationDisplay.autoPanMode = autoPanMode
+            }
+        }
+        
+        /// The status text to display to the user.
+        @Published var statusText: String = defaultStatus
+        
+        /// A map with a navigation basemap style.
+        let map = Map(basemapStyle: .arcGISNavigation)
+        
+        /// The map's location display.
+        let locationDisplay = LocationDisplay()
+        
+        /// The default status to display when not navigating.
+        static let defaultStatus = "Directions are shown here."
+        
+        /// The route task.
+        private let routeTask = RouteTask(url: .routeTask)
+        
+        /// The route result.
+        private var routeResult: RouteResult?
+        
+        /// The route tracker.
+        private var routeTracker: RouteTracker!
+        
+        /// The directions for the route.
+        private var directions: [DirectionManeuver] = []
+        
+        /// An AVSpeechSynthesizer for text to speech.
+        private let speechSynthesizer = AVSpeechSynthesizer()
+        
+        /// The graphics overlay for the stops.
+        private let stopGraphicsOverlay: GraphicsOverlay
+        
+        /// The graphics overlay for the route.
+        private let routeGraphicsOverlay: GraphicsOverlay
+        
+        /// Formats the time remaining for display.
+        private let timeFormatter: DateComponentsFormatter = {
```

Comment:
This formatter is still needed for iOS 15: https://stackoverflow.com/a/72945284/14369688